src/Config/HotReloadEventHandler.cs

```diff
@@ -0,0 +1,97 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Azure.DataApiBuilder.Config
+{
+    public class HotReloadEventHandler<TEventArgs> where TEventArgs : CustomEventArgs
+    {
+        public event EventHandler<TEventArgs>? QueryManagerFactory_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? MetadataProviderFactory_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? QueryEngineFactory_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? MutationEngineFactory_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? QueryExecutor_ConfigChangeEventOccurred;
+        public event EventHandler<TEventArgs>? MsSqlQueryExecutor_ConfigChangeEventOccurred;
```

Comment:
When do you instantiate all these event handlers? i.e. new EventHandler<TEventArgs> ?

Reply:
This are instantiated through the Subscribe function. In the subscribing class's constructor we take the HotReloadEventHandler instance that we created on startup, and we invoke its subscribe function, then we pass along the callback function as the eventhandler that is then assigned. When the event is later triggered it is this callback function that is then triggered.